### ISSUE TITLE
[Desktop] Brave search terms sent to ISP

### ISSUE THREAD

**Issue by erotavlasme** [brave/brave-browser#10251]
Hi,
both firefox and chrome have [this vulnerability](https://github.com/samduy/blog/tree/master/firefox/privacy-leakage). Since brave is based on chromium, does it suffer about the same issue? There is an [unfixed bug](https://bugs.chromium.org/p/chromium/issues/detail?id=479620) on chromium.

**Comment by rebron** [comment]
cc: @fmarier Can you take a look? Not sure if search autocomplete disabled off is enough here or maybe not related.

**Comment by fmarier** [comment]
Yes, I reproduced the leak **with search autocomplete disabled** using the following test plan:

1. Start Brave and wait 30 seconds for things to settle.
2. Run `sudo tcpdump udp port 53` in a terminal.
3. Type "supersecret" in the URL bar and press Enter.

You'll see something like this 

    19:24:38.402826 IP foobar.40262 > _gateway.domain: 266+ [1au] A? supersecret.lan. (44)
    19:24:38.405112 IP _gateway.domain > foobar.40262: 266 ServFail 0/0/1 (44)
    19:24:38.405227 IP foobar.40262 > _gateway.domain: 266+ A? supersecret.lan. (33)
    19:24:38.408063 IP _gateway.domain > foobar.40262: 266 ServFail 0/0/0 (33)

with the important part being the two unencrypted DNS queries for `supersecret.lan`.

The `.lan` part comes from this entry in `/etc/resolv.conf`:

    search lan

**Comment by fmarier** [comment]
It seems like [there is work being done on the Chromium side](https://bugs.chromium.org/p/chromium/issues/detail?id=479620#c141) but ultimately, this is a nasty side effect of combining the URL box with the search box and then using heuristics to tell search queries apart from URLs. (Firefox used to have two boxes and while that's still an option there, it's not the default anymore.)

**Comment by diracdeltas** [comment]
FYI this was originally reported via hackerone and triaged to https://github.com/brave/brave-browser/issues/2248. going to close the old issue since this one has more details.

**Comment by bsclifton** [comment]
This issue hasn't had an update in a while - it's met the criteria for being stale. The issue is going to be closed for now. If this is still a problem, please leave a comment or re-open the issue. Thanks!
